OnionGalaxy/TwinlightSouls
# quick fix to avoid hierarchy position dependency

What's new:
This is a patch, until we figure it out how to solve/redo this.

I think there is a problem with the current pipepline/flow from the EnemyManager->EnemyController->Mushdoom->States.
Until now all the enemies were dependant from their position respective to World Manager, due to the current flow is not correct.
We need to remember, we can not assign player references in the Awake. 

HOW TO TEST IT:
First don't test this branch, go to your current branch, go to scene Level1.scene put the World Manager the last item, save scene, and reload the scene. Then click Play and see error.
Then come to this branch and do the same.
